modules/document_repository/jsx/uploadForm.js

```diff
@@ -0,0 +1,292 @@
+              name="forSite"
+              label="Site"
+              placeHolder="Search for site"
+              options={this.state.Data.sites}
+              strictSearch={true}
+              onUserInput={this.setFormData}
+              ref="forSite"
+              required={true}
+              value={this.state.formData.forSite}
+            />
+            <SelectElement
+              name="instrument"
+              label="Instrument"
+              options={this.state.Data.instruments}
+              onUserInput={this.setFormData}
+              ref="instrument"
+              value={this.state.formData.instrument}
+            />
+            <TextboxElement
+              name="pscid"
+              label="PSCID"
+              onUserInput={this.setFormData}
+              ref="pscid"
+              value={this.state.formData.pscid}
+            />
+            <TextboxElement
+              name="visitLabel"
+              label="Visit Label"
+              onUserInput={this.setFormData}
+              ref="visitLabel"
+              value={this.state.formData.visitLabel}
+            />
+            <TextboxElement
+              name="version"
+              label="Version"
+              onUserInput={this.setFormData}
+              ref="version"
+              value={this.state.formData.version}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              ref="comments"
+              value={this.state.formData.comments}
+            />
+            <FileElement
+              name="file"
+              id="docUploadEl"
+              onUserInput={this.setFormData}
+              ref="file"
+              label="File to upload"
+              required={true}
+              value={this.state.formData.file}
+            />
+            <ButtonElement label="Upload File"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+
+    let formData = this.state.formData;
+    let formRefs = this.refs;
+    // let docFiles = this.state.Data.docFiles ? this.state.Data.docFiles : [];
+
+    // Validate the form
+    if (!this.isValidForm(formRefs, formData)) {
+      return;
+    }
+      this.uploadFile();
+  }
+  /*
+   * Uploads the file to the server
+   */
+  uploadFile() {
+    // Set form data and upload the media file
+    let formData = this.state.formData;
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObj.append(key, formData[key]);
+      }
+    }
+    $.ajax({
```

Comment:
we're trying to move towards using the standard javascript fetch API instead of jQuery. @HenriRabalais @davidblader @zaliqarosli can one of you help @kongtiaowang figure out how to do that here?